.github/workflows/test.yml

```diff
@@ -0,0 +1,51 @@
+name: Test
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+        os: [ubuntu-18.04, ubuntu-16.04, macos-latest]
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Ubuntu cache
+        uses: actions/cache@v1
+        if: startsWith(matrix.os, 'ubuntu')
+        with:
+          path: ~/.cache/pip
+          key:
+            ${{ matrix.os }}-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
+          restore-keys: |
+            ${{ matrix.os }}-${{ matrix.python-version }}-
+
+      - name: macOS cache
+        uses: actions/cache@v1
+        if: startsWith(matrix.os, 'macOS')
+        with:
+          path: ~/Library/Caches/pip
+          key:
+            ${{ matrix.os }}-${{ matrix.python-version }}-${{ hashFiles('**/pyproject.toml') }}
+          restore-keys: |
+            ${{ matrix.os }}-${{ matrix.python-version }}-
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v1
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install --upgrade poetry
+          poetry install
+
+      - name: Test
+        shell: bash
```

Comment:
why specify bash here?

Reply:
There are different defaults for different platforms:

https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

Reply:
True, but this runs on macOS and ubuntu, which both default to `bash`.

Reply:
Current releases of macOS default to `zsh`, not `bash` so I approve of the _explicit is better than implicit_ approach used here.

Reply:
it's a nitpick, so i feel fairly indifferent about it.

Reply:
Yeah, Windows would be a great addition for the future as AppVeyor is really slow with a single concurrent job:

> Not included:
> ...
> * Windows. Omitted for now, the poetry install was failing. As this issue is about replacing Travis and not AppVeyor, that can wait.